client/src/screens/groups/GroupDetails.js

```diff
@@ -3,252 +3,89 @@ import React, { Component } from 'react';
-    id: PropTypes.number.isRequired,
-    name: PropTypes.string.isRequired,
-  }),
-};
-
-const ScheduleDisplay = (props) => {
-  const { id, name, details, startTime, endTime } = props.schedule;
-  let timeText = '';
-  if (startTime === 0 && endTime === distantFuture) {
-    timeText = 'Perpetual Schedule';
-  } else {
-    const startText = moment.unix(startTime).format('DD/MM/YY, HH:mm:ss');
-    const endText = moment.unix(endTime).format('DD/MM/YY, HH:mm:ss');
-    timeText = `${startText} - ${endText}`;
-  }
-  return (
-    <TouchableHighlight key={id}>
-      <View style={styles.scheduleContainer}>
-        <Icon name="calendar" size={24} color="blue" />
-        <View style={styles.scheduleTextContainer}>
-          <View style={styles.groupTitleContainer}>
-            <Text style={styles.scheduleName} numberOfLines={1}>{name} - {timeText}</Text>
-          </View>
-          <Text style={styles.scheduleText} numberOfLines={1}>{details}</Text>
-        </View>
-      </View>
-    </TouchableHighlight>
-  );
-};
-ScheduleDisplay.propTypes = {
-  schedule: PropTypes.shape({
-    id: PropTypes.number.isRequired,
-    name: PropTypes.string.isRequired,
-    details: PropTypes.string,
-    startTime: PropTypes.number.isRequired,
-    endTime: PropTypes.number.isRequired,
-  }),
+const GroupDetailsList = ({ items, render, networkStatus, refetch }) => (
+  <Container>
+    <FlatList
+      data={items}
+      ListHeaderComponent={() =>
+        (!items.length ? (
+          <Center>
+            <Text>Nothing here</Text>
+          </Center>
+        ) : null)}
+      keyExtractor={item => item.id}
+      renderItem={({ item }) =>
+        // eslint-disable-next-line no-nested-ternary
+      (render === 'UserListItem' ? (
+        <UserListItem
+          user={item}
+          // TODO: Deep link
+          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to user details')}
+        />
+      ) : render === 'ScheduleListItem' ? (
+        <ScheduleListItem
+          schedule={item}
+          // TODO: Deep link
+          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to schedule details')}
+        />
+      ) : render === 'EventListItem' && (
+        <EventListItem
+          event={item}
+          // TODO: Deep link
+          onPress={() => Alert.alert('PLACEHOLDER', 'Deeplink to event details')}
+        />
+      ))
+      }
+      refreshing={networkStatus === 4}
+      onRefresh={refetch}
+    />
+  </Container>
+);
+GroupDetailsList.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
+  items: PropTypes.array.isRequired,
+  render: PropTypes.string.isRequired,
+  networkStatus: PropTypes.number.isRequired,
+  refetch: PropTypes.func,
 };
 
 class GroupDetails extends Component {
   static navigationOptions = () => ({
     title: 'Group Details',
   });
 
+  state = {
+    selectedIndex: 0,
+  };
+
+  componentDidMount() {
+    this.props.navigation.setParams({ navIconJoinLeave: 'sign-out' });
```

Comment:
what does this do? add a comment explaining it

Reply:
this should not be in there. nice catch